### ISSUE TITLE
SvrProcess must handle mandatory error on Process Parameters

### ISSUE THREAD

**Issue by yamelsenih** [adempiere/adempiere#325]
Is nice to have that in SvrProcess the mandatory process parameters validation.
Also is good have a method for get parameter, example:

```java
  protected String doIt () throws Exception {
     boolean isSOTrx = getValueAsBoolean("IsSOTrx");
     int C_Invoice_ID = getValueAsInt("C_Invoice_ID");
  }

**Comment by e-Evolution** [comment]
Hi Yamel I, think that is best remove of underscore getParameterToAsTimestamp(String parameterName)

This is should a best way when using the IDE, the other hand the java the Upper Case is used to separate world

kind regards
Victor

**Comment by yamelsenih** [comment]
Hi @e-Evolution i put it like **getParameter_ToAsTimestamp(String parameterName)** because the parameter name also is ending in **_To**. But it is true, i will change the underscore in **To** parameter.

**Comment by mckayERP** [comment]
Problem with the code to validate process parameters does not allow zero values in IDs.  For the Role Access Update, the AD_Client_ID for System is zero.  The current code does not allow the Role Access Update process to run for System.

**Comment by yamelsenih** [comment]
Hi @mckayERP thanks you for it, I will see it for change

**Comment by mckayERP** [comment]
Morning Yamel.  Perhaps the getParameterAsInt function can return -1 if the value is null rather than 0.  Then you can test for < 0 when its an ID/Lookup.  Cheers.

**Comment by yamelsenih** [comment]
Hi @mckayERP the change has been, you can test it?
Thank you a lot for find it.

**Comment by mckayERP** [comment]
Tested OK. Did not test null values - current code will return null BigDecimal parameters as zero which could be interpreted as valid data.  Modified the MigrationLoader to add mandatory parameters and switched to class names instead of Process IDs.  Depends on #363.
